src/main/java/com/airlines/dao/impl/CrewMemberDaoImpl.java

```diff
@@ -0,0 +1,107 @@
+                    "WHERE id = ?;";
+    private final static String SELECT_CREW_MEMBER_BY_ID_SQL = "SELECT * FROM crew_member WHERE id = ?;";
+
+    private final DbConnector dbConnector;
+
+    public CrewMemberDaoImpl(DbConnector dbConnector) {
+        this.dbConnector = dbConnector;
+    }
+
+    @Override
+    public void save(CrewMember crewMember) {
+        Objects.requireNonNull(crewMember);
+
+        try (Connection connection = dbConnector.getDBConnection()) {
+
+            PreparedStatement insertStatement = connection.prepareStatement(INSERT_CREW_MEMBER_SQL);
+            insertStatement.setInt(1, crewMember.getId());
+            insertStatement.setString(2, crewMember.getFirstName());
+            insertStatement.setString(3, crewMember.getLastName());
+            insertStatement.setString(4, crewMember.getPosition().toString());
+            insertStatement.setDate(5, Date.valueOf(crewMember.getBirthday()));
+            insertStatement.setString(6, crewMember.getCitizenship().toString());
+            insertStatement.setInt(7, crewMember.getCrewId());
+            insertStatement.executeUpdate();
+
+        } catch (SQLException e) {
+            throw new DaoOperationException("Error saving crew member" + crewMember, e);
+        }
+    }
+
+    @Override
+    public void update(CrewMember crewMember) {
+        Objects.requireNonNull(crewMember);
+
+        if (crewMember.getId() == 0) {
+            throw new DaoOperationException("Cannot find a product without ID");
+        }
+
+        try (Connection connection = dbConnector.getDBConnection()) {
+
+            PreparedStatement updateStatement = connection.prepareStatement(UPDATE_CREW_MEMBER_SQL);
+            updateStatement.setString(1, crewMember.getFirstName());
+            updateStatement.setString(2, crewMember.getLastName());
+            updateStatement.setString(3, crewMember.getPosition().toString());
+            updateStatement.setDate(4, Date.valueOf(crewMember.getBirthday()));
+            updateStatement.setString(5, crewMember.getCitizenship().toString());
+            updateStatement.setInt(6, crewMember.getCrewId());
+            updateStatement.setInt(7, crewMember.getId());
+            int rowsAffected = updateStatement.executeUpdate();
+
+            if (rowsAffected == 0) {
+                throw new DaoOperationException(String.format("Crew member with id = %d does not exist", crewMember.getId()));
+            }
+        } catch (SQLException e) {
+            throw new DaoOperationException(String.format("Crew member with id = %d does not exist", crewMember.getId()), e);
+        }
+    }
+
+    @Override
+    public CrewMember findById(int id) {
+        try (Connection connection = dbConnector.getDBConnection()) {
+
+            PreparedStatement selectByIdStatement = connection.prepareStatement(SELECT_CREW_MEMBER_BY_ID_SQL);
```

Comment:
PreparedStatement also resource that use connection to database and its also marked as Autoclosable, so it's better move to try with resources decalring try ( Connection .... ; PreparedStatement ...

Reply:
done in https://github.com/pielena/airlines-jdbc-app/commit/21a3eec6b8bb439efba0fe86562c5f31b291e47e